param/__init__.py

```diff
@@ -1112,33 +1097,40 @@ class Composite(Parameter):
     attributes.
     """
 
-    __slots__=['attribs','objtype']
+    __slots__ = ['attribs', 'objtype']
 
-    def __init__(self,attribs=None,**kw):
+    def __init__(self, attribs=None, **kw):
         if attribs is None:
             attribs = []
-        super(Composite,self).__init__(default=None,**kw)
+        super(Composite, self).__init__(default=None, **kw)
         self.attribs = attribs
 
-    def __get__(self,obj,objtype):
+    def __get__(self, obj, objtype):
         """
         Return the values of all the attribs, as a list.
         """
         if obj is None:
-            return [getattr(objtype,a) for a in self.attribs]
+            return [getattr(objtype, a) for a in self.attribs]
         else:
-            return [getattr(obj,a) for a in self.attribs]
+            return [getattr(obj, a) for a in self.attribs]
+
+    def _validate_attribs(self, val, attribs):
+        if len(val) == len(attribs):
+            return
+        raise ValueError("Compound parameter %r got the wrong number "
+                         "of values (needed %d, but got %d)." %
+                         (self.name, len(attribs), len(val)))
 
     def _validate(self, val):
-        assert len(val) == len(self.attribs),"Compound parameter '%s' got the wrong number of values (needed %d, but got %d)." % (self.name,len(self.attribs),len(val))
```

Comment:
Guess I've never seen this assert raise an exception so I guess it is safe to remove...

Reply:
Not removed, just moved to `_validate_attribs` as a proper `ValueError`